apps/aehttp/test/aehttp_hyperchains_SUITE.erl

```diff
@@ -1735,31 +1735,35 @@ produce_cc_blocks(Config, BlocksCnt, ParentProduce) ->
 produce_to_cc_height(Config, TopHeight, GoalHeight, ParentProduce) ->
     NodeNames = [ Name || {_, Name, _} <- ?config(nodes, Config) ],
     BlocksNeeded = GoalHeight - TopHeight,
-    case  BlocksNeeded > 0 of
+    case BlocksNeeded > 0 of
         false ->
             TopHeight;
         true ->
             NewParentProduce =
                 case ParentProduce of
-                    [{CH, PBs} | PRest ]  when CH == TopHeight+1 ->
+                    [{CH, PBs} | PRest ] when CH == TopHeight+1 ->
                         mine_key_blocks(?PARENT_CHAIN_NODE_NAME, PBs),
                         PRest;
                     PP -> PP
                 end,
-            KeyBlock =
-                case rpc:call(hd(NodeNames), aec_tx_pool, peek, [infinity]) of
-                    {ok, []} ->
-                         {ok, [{N, Block}]} = mine_cc_blocks(NodeNames, 1),
-                         ct:log("CC ~p mined block: ~p", [N, Block]),
-                         Block;
-                    {ok, _Txs} ->
-                         {ok, [{N1, MB}, {N2, KB}]} = mine_cc_blocks(NodeNames, 2),
-                         ?assertEqual(key, aec_blocks:type(KB)),
-                         ?assertEqual(micro, aec_blocks:type(MB)),
-                         ct:log("CC ~p mined micro block: ~p", [N1, MB]),
-                         ct:log("CC ~p mined key block:   ~p", [N2, KB]),
-                         KB
-                end,
+
+            %% TODO: add some assertions when we expect an MB (and not)!
+            {ok, _Txs} = rpc:call(hd(NodeNames), aec_tx_pool, peek, [infinity]),
+
+            %% This will mine 1 key-block (and 0 or 1 micro-blocks)
+            {ok, Blocks} = mine_cc_blocks(NodeNames, 1),
+
+            {Node, KeyBlock} = lists:last(Blocks),
+            case Blocks of
+                [{Node, MB}, _] ->
+                    ?assertEqual(micro, aec_blocks:type(MB)),
+                    ct:log("CC ~p produced micro-block: ~p", [Node, MB]);
+                [_] ->
+                    ok
```

Comment:
What is the expectation to see here? In the HC suite, `verify_fees`, I'm seeing two micro block (candidates?), and FAIL with `case_clause`. I'm guessing the first one (single spend to/from a non-validator adress i.e. not ALICE, LISA or BOB) has something to do with the actual test case, and the second one is the pinning "notarisation" spend to/from a validator. It looks like this should match the case, but who am I to say? BEAM obviously don't think so...

Reply:
There should never be two MBs for one "generation"/block. The check is correct.

Reply:
Houston, we have a problem then.